users/views.py

```diff
@@ -0,0 +1,188 @@
+            response["message"] = "User does not exist"
+
+        return Response(response)
+
+
+class UserAPIView(APIView):
+    """
+    User Management API
+    post to add user
+    put to login user
+    patch to update user
+    delete to remove user
+    """
+
+    def post(self, request):
+        """
+        This function adds a user to db
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        response = {}
+        try:
+            name = request.POST["name"]
+            email = request.POST["email"].lower()
+            password = request.POST["password"]
+            validate_password(password)
+            validate_email(email)
+            unique_email = not User.objects.filter(email=email).exists()
+            valid_name = name not in ("", " ")
+            if valid_name and unique_email:
+                User.objects.create_user(email=email, password=password, name=name)
+                response["message"] = "Successfully created user."
+                response["code"] = status.HTTP_201_CREATED
+            else:
+                response["message"] = "Email already exists."
+                if not valid_name:
+                    response["message"] = "Name can not be blank."
+                response["code"] = status.HTTP_400_BAD_REQUEST
+        except KeyError:
+            response["message"] = "Required fields don't exist."
+            response["code"] = status.HTTP_400_BAD_REQUEST
+        except IntegrityError as integrity_error:
+            response["message"] = str(integrity_error)
+            response["code"] = status.HTTP_409_CONFLICT
+        except ValidationError as validation_error:
+            response["message"] = " ".join(validation_error)
+            response["code"] = status.HTTP_400_BAD_REQUEST
+
+        return Response(response)
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def patch(self, request):
+        """
+        This function updates data of existing user
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        try:
+            user = request.user
+            if "email" in request.POST:
+                email = request.POST["email"]
+                validate_email(email)
+                if not User.objects.filter(email=email).exists():
+                    user.email = request.POST["email"]
+            if "name" in request.POST:
+                name = request.POST["name"]
+                if name not in ["", " "]:
+                    user.name = name
+                else:
+                    return Response(
+                        {
+                            "message": "Name is not valid",
+                            "code": status.HTTP_400_BAD_REQUEST,
```

Comment:
How about we use the validation through the Serializer by passing the payload into that and then calling `is_valid()` that would be a common point of reference for us